src/span.h

```diff
@@ -206,4 +208,20 @@ T& SpanPopBack(Span<T>& span)
     return back;
 }
 
+//! Mimick C++20's std::as_bytes (with unsigned char instead of std::byte)
+template<typename T>
+Span<const unsigned char> AsUnsignedChars(Span<T> in) { return {reinterpret_cast<const unsigned char*>(in.data()), in.size_bytes()}; }
```

Comment:
Agree, that seems like both a more convenient and slightly safer construction. I've implemented it slightly differently so that it remains constexpr, and supports everything `MakeSpan` does (including passing temporaries, and non-const return when appropriate).

> In commit "Make uint256 Span-convertible by adding ::data()" (5d1f1d9)

I think you mean a different commit?

> My concern is that providing this function might make it too easy to write code that looks like it is hashing `vector<uint256>` or `Optional<uint256>` or `uint256*` data, but actually hashing the outer pointer or container objects instead.

I don't think that would be a large concern. Those data types don't have `data()` and `size()` methods, so no implicit conversion to `Span` can happen. You can of course explicitly construct one, but you'd need to use something like `AsUnsignedChars(Span<std::vector<uint256>(&vec, 1))`, which is hopefully obviously accessing the vector's byte representation instead.